Model/Utilities/WebKitHandler.swift

```diff
@@ -53,37 +48,64 @@ final class KiwixURLSchemeHandler: NSObject, WKURLSchemeHandler {
         }
     }
 
+    func webView(_ webView: WKWebView, stop urlSchemeTask: WKURLSchemeTask) {
+        stopFor(urlSchemeTask.hash)
+    }
+
+    func didFailProvisionalNavigation() {
+        stopAll()
+    }
+
     func webView(_ webView: WKWebView, start urlSchemeTask: WKURLSchemeTask) {
         let hash = urlSchemeTask.hash
         guard isStartedFor(hash) == false else { return }
         startFor(hash)
 
         queue.async { [weak self] in
-            guard let url = urlSchemeTask.request.url, url.isKiwixURL else {
+            let request = urlSchemeTask.request
+            guard let url = request.url, url.isKiwixURL else {
                 urlSchemeTask.didFailWithError(URLError(.unsupportedURL))
                 self?.stopFor(hash)
                 return
             }
-            guard let content = ZimFileService.shared.getURLContent(url: url) else {
+            var start: UInt = 0
+            var end: UInt = 0
+            var isVideo = false
+            if let mimeType = ZimFileService.shared.getMimeType(url: url) {
+                isVideo = mimeType.contains("video")
+                if isVideo {
+                    if let range = request.allHTTPHeaderFields?["Range"] as? String {
+                        let parts = range.components(separatedBy: ["=", "-"])
+                        guard parts.count > 1, let rangeStart = UInt(parts[1]) else {
+                            self?.sendHTTP400Response(urlSchemeTask, url: url)
+                            return
+                        }
+                        let rangeEnd = parts.count == 3 ? UInt(parts[2]) ?? 0 : 0
+                        start = rangeStart
+                        end = rangeEnd
+                    }
+                }
+            }
+            guard let content = ZimFileService.shared.getURLContent(url: url, start: start, end: end) else {
                 self?.sendHTTP404Response(urlSchemeTask, url: url)
                 self?.stopFor(hash)
                 return
             }
-            self?.sendHTTP200Response(urlSchemeTask, url: url, content: content)
+            if isVideo {
+                self?.sendHTTP206Response(urlSchemeTask, url: url, content: content, start: start, end: end)
```

Comment:
we only apply the range responses to video type of contents

Reply:
_Only apply to_ but also _Apply to ALL videos_…